src/catalog/src/durable.rs

```diff
@@ -311,6 +314,100 @@ pub trait DurableCatalogState: ReadOnlyDurableCatalogState {
         let id = id.into_element();
         Ok(ReplicaId::User(id))
     }
+
+    /// Migrates storage stash data from `other` to `self`.
+    async fn migrate_storage_state_from(
+        &mut self,
+        other: &mut Box<dyn DurableCatalogState>,
+    ) -> Result<(), CatalogError> {
```

Comment:
This signature is a bit confusing to me. I thought we would need to migrate the storage data from the storage controller's stash to the catalog. This suggests that we're migrating the storage data from one catalog implementation to another catalog implementation. How did the data get into one implementation but not the other?

Reply:
I think you might be blurring the abstraction boundaries of the `DurableCatalogState` and the `Stash`.

Reply:
Yeah--this slotted in more naturally just using `Stash`.